File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = []
    cnt = 0
    for k in moves:
        for i in range(len(board)):
            if board[i][k-1] !=0:
                answer.append(board[i][k-1])
                board[i][k-1] =0
                break
        while len(answer) >=2 and answer[-1] == answer[-2]:
            answer.pop()
            answer.pop()
            cnt+=2
            
    
    return cnt

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        
        Stack<Integer> st = new Stack<>();

        
        for(int m : moves){
            for(int i = 0; i < board.length;i++){
                if(board[i][m-1]!=0){
                    st.add(board[i][m-1]);
                    board[i][m-1] =0;
                    break;
                }
            }
            while((st.size()>2 || st.size()==2) && st.elementAt(st.size()-2)==st.peek()){
                st.pop();
                st.pop();
                answer+=2;
            }
        }
        return answer;
    }
}